src/core/thread/router_table.cpp

```diff
@@ -596,6 +596,25 @@ void RouterTable::UpdateRoutes(const Mle::RouteTlv &aRouteTlv, uint8_t aNeighbor
                 neighbor->SetLinkQualityOut(linkQuality);
                 SignalTableChanged();
             }
+
+            // If the `aRouteTlv` indicates that the neighboring
+            // router claims to have no link to us (by setting its
+            // `GetLinkQualityOut()` towards us as `kLinkQuality0`),
+            // and we have previously established a link with it, and
+            // our two-way link quality to this router is at least
+            // `kLinkQuality2`, we schedule a unicast Advertisement to
+            // be sent to this neighbor. This helps expedite recovery
+            // from any temporary router link quality mismatch.
+            // Otherwise, the neighboring router will continue to
+            // advertise that it has no link to us until our next
+            // trickle timer-triggered Advertisement transmission
+            // (which can be up to 32 seconds later).
+
+            if (neighbor->IsStateValid() && (aRouteTlv.GetLinkQualityOut(index) == kLinkQuality0) &&
```

Comment:
Should we also check here whether a unicast MLE Advertisement to that neighbor is already scheduled?
It would seem overkill to send / schedule multiple.   If already scheduled, don't re-schedule it.

Reply:
Yes. The `DelayedSender` module (which is used by `ScheduleUnicastAdvertisementTo()`) will do this for us already. It will ignore a new schedule request for the same destination if one is already scheduled:

```c++
void Mle::DelayedSender::ScheduleAdvertisement(const Ip6::Address &aDestination, uint16_t aDelay)
{
    VerifyOrExit(!HasMatchingSchedule(kTypeAdvertisement, aDestination));
    AddSchedule(kTypeAdvertisement, aDestination, aDelay, nullptr, 0);

exit:
    return;
}
```

One reason for adding `DelayedSender` was to apply the correct logic per MLE message type. For some message types, we replace the old scheduled message with a new one (e.g., `ParentResponse` or `DataResponse`). For some, we ignore the new request as the old one is fine and valid,  (e.g., `LinkRequest`. `ChildUpdate`, ...).